localstack/services/awslambda/invocation/version_manager.py

```diff
@@ -527,14 +537,20 @@ def process_event_destinations(
             isinstance(invocation_result, InvocationError)
             and self.function_version.config.dead_letter_arn
         ):
-            dead_letter_queue._send_to_dead_letter_queue(
-                source_arn=self.function_arn,
-                dlq_arn=self.function_version.config.dead_letter_arn,
-                event=json.loads(to_str(original_payload)),
-                error=InvocationException(
-                    message="hi", result=to_str(invocation_result.payload)
-                ),  # TODO: check message
-            )
+            try:
+                dead_letter_queue._send_to_dead_letter_queue(
+                    source_arn=self.function_arn,
+                    dlq_arn=self.function_version.config.dead_letter_arn,
+                    event=json.loads(to_str(original_payload)),
+                    error=InvocationException(
+                        message="hi", result=to_str(invocation_result.payload)
+                    ),  # TODO: check message
+                    role=self.function_version.config.role,
+                )
+            except Exception as e:
+                LOG.warning(
```

Comment:
:+1: 

Definitely good to start catching a few more of these exceptions and properly log them. Especially for things started asynchronously in a separate thread.

Reply:
Yes. I had some problems with the logging, as it used too many permissions, and it would just crash the whole thread, not good :)